src/udev/udevadm-control.c

```diff
@@ -109,17 +109,36 @@ int control_main(int argc, char *argv[], void *userdata) {
                                 return log_error_errno(r, "Failed to send exit request: %m");
                         break;
                 case 'l':
-                        r = log_level_from_string(optarg);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to parse log level '%s': %m", optarg);
-
-                        r = json_build(&v, JSON_BUILD_OBJECT(JSON_BUILD_PAIR("log-level", JSON_BUILD_INTEGER(r))));
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to build json object: %m");
-
-                        r = udev_varlink_call(link, "io.systemd.udev.SetLogLevel", v, NULL);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to send request to set log level: %m");
+                        if (optarg) {
+                            r = log_level_from_string(optarg);
+                            if (r < 0)
+                                    return log_error_errno(r, "Failed to parse log level '%s': %m", optarg);
+
+                            r = json_build(&v, JSON_BUILD_OBJECT(JSON_BUILD_PAIR("log-level", JSON_BUILD_INTEGER(r))));
+                            if (r < 0)
+                                    return log_error_errno(r, "Failed to build json object: %m");
+
+                            r = udev_varlink_call(link, "io.systemd.udev.SetLogLevel", v, NULL);
+                            if (r < 0)
+                                    return log_error_errno(r, "Failed to send request to set log level: %m");
+                        } else {
+                            JsonVariant *log_level;
+                            _cleanup_free_ char *log_level_str = NULL;
+
+                            r = udev_varlink_call(link, "io.systemd.udev.GetLogLevel", NULL, &v);
```

Comment:
afaics `v` is leaked if you do `udevadm control --log-level --log-level --log-level`, no?